### PR TITLE
fix: not check nested call aggregate functions 

### LINKED ISSUE #13233: bug: not check nested call aggregate functions
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v1.2.154-nightly

### What's Wrong?

not check nested call aggregate functions

### How to Reproduce?

```sql
mysql> create table test(a int, b date);
Query OK, 0 rows affected (0.25 sec)

mysql> select arg_min(DATE_ADD(MINUTE, COUNT(*), b), a) FROM test;
ERROR 1105 (HY000): BadArguments. Code: 1006, Text = Unable to get field named "18446744073709551615". Valid fields: ["0", "1"].
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by zenus**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #13233

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13372)

### CODE REVIEW

File: src/query/sql/src/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -951,6 +951,23 @@ impl<'a> TypeChecker<'a> {
...
+                        return Err(ErrorCode::SemanticError(
+                            "aggregate function calls cannot be nested".to_string(),
+                        )
+                        .set_span(*span));
+                    }
+                }
+
                 let agg_func = AggregateCountFunction::try_create("", vec![], vec![])?;
 
                 let (new_agg_func, data_type) = (
```
Comment (b41sh) [inline src/query/sql/src/planner/semantic/type_check.rs:973]: we can rewrite this as
```rust
let (new_agg_func, data_type) = self
    .resolve_aggregate_function(*span, "count", expr, false, &[], &[])
    .await?;
```
Aggregate function nested call will be checked in `resolve_aggregate_function`. There's no need to double-check here.

Comment (zenus) [reply src/query/sql/src/planner/semantic/type_check.rs]: @b41sh  thanks

File: tests/sqllogictests/suites/query/02_function/02_0012_function_datetimes [hunk 2]
```diff
@@ -1008,3 +1008,18 @@ query I
 SELECT to_week_of_year('2016-01-02T23:39:20.123-07:00');
 ----
 53
+
+statement ok
+drop table if exists tc
```
Comment (b41sh) [inline tests/sqllogictests/suites/query/02_function/02_0012_function_datetimes:1013]: better to move these tests to `tests/sqllogictests/suites/query/aggregate.test`

**Review by b41sh — APPROVED** [review]
LGTM

**Pull request merged**